Remove animations when it's disabled in system settings

stevenxxiu opened the pull request:
Closes #185.

stevenxxiu committed:
feat: disable logs tab transition

stevenxxiu committed:
feat: disable logs horizontal overscroll

stevenxxiu committed:
feat: disable fragment transitions

stevenxxiu committed:
feat: disable repository module tab transition

JingMatrix commented on app/src/main/java/org/lsposed/manager/util/AccessibilityUtils.java:
We don't need to add this `LSPosed` header. It is redundant, no copyright information is necessary. 
Please remove it.

JingMatrix commented on app/src/main/java/org/lsposed/manager/ui/fragment/RepoItemFragment.java:
Could you please explain the parameter that you pass to the argument `autoRefresh`?

https://developer.android.com/reference/com/google/android/material/tabs/TabLayoutMediator#TabLayoutMediator(com.google.android.material.tabs.TabLayout,%20androidx.viewpager2.widget.ViewPager2,%20boolean,%20boolean,%20com.google.android.material.tabs.TabLayoutMediator.TabConfigurationStrategy)

stevenxxiu replied:
Sure, I added a comment. I couldn't find docs except for some old commit in <https://android.googlesource.com/platform/frameworks/support/+/9bc1d32ac0080e030959f86e46041efd28e62ad4/viewpager2/integration-tests/testapp/src/main/java/androidx/viewpager2/integration/testapp/TabLayoutMediator.java#91>. I just copied that:

> update the tabs automatically when the data set of the view pager's adapter changes

It's set to `true` by default if not set. Seems like this doesn't affect the animation anyway. I set it back to `true` now. Some *StackOverflow* answer just set it to `false`, so I didn't think much about it.

The argument has some controversy too at <https://github.com/material-components/material-components-android/issues/887>.

JingMatrix commented on app/src/main/java/org/lsposed/manager/ui/fragment/LogsFragment.java:
```diff
@@ -109,7 +110,15 @@ public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup c
+        new TabLayoutMediator(
+            binding.tabLayout,
+            binding.viewPager,
+            isAnimationEnabled,
```
Please update this part about the autoRefresh parameter as well.

stevenxxiu replied:
I did just now, did you mean something else?

JingMatrix replied:
Nope, you forgot changing it in this file. There were two places to change, and I commented on only one to avoid repeating myself.

JingMatrix committed:
Use default NavOptions constructor to clear animations

To navigate without animations, the value for arguments of type @AnimatorRes should be -1.

JingMatrix commented:
If possible, please consider hiding the component ~~`CircularProgressIndicator`~~ `SwipeRefreshLayout`.
It is inconsistent to have a progress indicator with other animations disabled.

stevenxxiu commented:
> If possible, please consider hiding the component ~`CircularProgressIndicator`~ `SwipeRefreshLayout`. It is inconsistent to have a progress indicator with other animations disabled.

If `SwipeRefreshLayout` is removed, then is there a way to tell when the refresh finishes? Otherwise it could be something that's informative. I don't mind spinner loading animations for that reason.

JingMatrix commented:
Okay for that.

I wonder if if the `smoothScroll` parameter of [TabLayoutMediator](https://developer.android.com/reference/com/google/android/material/tabs/TabLayoutMediator#TabLayoutMediator(com.google.android.material.tabs.TabLayout,%20androidx.viewpager2.widget.ViewPager2,%20boolean,%20com.google.android.material.tabs.TabLayoutMediator.TabConfigurationStrategy)) really creates animations.
I cannot tell the difference from my device.

stevenxxiu commented:
> I wonder if if the `smoothScroll` parameter of [TabLayoutMediator](https://developer.android.com/reference/com/google/android/material/tabs/TabLayoutMediator#TabLayoutMediator(com.google.android.material.tabs.TabLayout,%20androidx.viewpager2.widget.ViewPager2,%20boolean,%20com.google.android.material.tabs.TabLayoutMediator.TabConfigurationStrategy)) really creates animations. I cannot tell the difference from my device.

It should remove the sliding animation that shows up when you switch tabs in the `Logs` panel or in  a `Repository` panel's module. 

I tried this in an emulator before, and following your comments tried it on my real phone. It works there too.

Are you sure you disabled all animations in system settings? "Animation duration scale", "Window animation scale", "Transition animation scale".
